src/main/java/org/dave/compactmachines3/item/ItemPersonalShrinkingDevice.java

```diff
@@ -56,20 +54,20 @@ public void addInformation(ItemStack stack, @Nullable World worldIn, List<String
     public ActionResult<ItemStack> onItemRightClick(World world, EntityPlayer player, EnumHand hand) {
         ItemStack stack = player.getHeldItem(hand);
         if(hand == EnumHand.OFF_HAND) {
-            return new ActionResult(EnumActionResult.FAIL, stack);
+            return new ActionResult<>(EnumActionResult.FAIL, stack);
         }
 
         if(world.provider.getDimension() != ConfigurationHandler.Settings.dimensionId) {
             player.openGui(CompactMachines3.instance, GuiIds.PSD_GUIDE.ordinal(), world, (int) player.posX, (int) player.posY, (int) player.posZ);
-            return new ActionResult(EnumActionResult.SUCCESS, stack);
+            return new ActionResult<>(EnumActionResult.SUCCESS, stack);
         }
 
         if(!world.isRemote && player instanceof EntityPlayerMP) {
             EntityPlayerMP serverPlayer = (EntityPlayerMP)player;
 
             if(player.isSneaking()) {
                 int id = StructureTools.getIdForPos(player.getPosition());
-                Vec3d pos = player.getPositionVector();
+                Vec3d2f pos = new Vec3d2f(player);
```

Comment:
We should consider not making a new class here and just add a Vec2f field instead